### PR TITLE
Updated javadocs for GeneralIndexModule.

### PR DESCRIPTION
**Pull request by bweick**
# Code Review Processes
## New Feature Review
Before submitting a pull request for new review, make sure the following is done:
* Design doc is created and posted here: [Insert Link]
* Code cleanliness and completeness is addressed via [guidelines](https://app.gitbook.com/@setprotocol-1/s/set/smart-contract-engineering/sc-code-review-process)

README Checks
- [] README has proper context for the reviewer to understand what the code includes, any important design considerations, and areas to pay more attention to

Code Checks
- [] Add explanatory comments. If there is complex code that requires specific context or understanding, note that in a comment
- [] Remove unncessary comments. Any comments that do not add additional context, information, etc. should be removed
- [] Add javadocs. 
- [] Scrub through the code for inconsistencies (e.g. removing extra spaces)
- [] Ensure there are not any .onlys in spec files

Broader Considerations
- [] Ensure variable, function and event naming is clear, consistent, and reflective for the scope of the code.
- [] Consider if certain pieces of logic should be placed in a different library, module

### CODE REVIEW

File: contracts/protocol/modules/GeneralIndexModule.sol [hunk 1]
```diff
@@ -299,9 +317,9 @@ contract GeneralIndexModule is ModuleBase, ReentrancyGuard {
     }
 
     /**
-     * ACCESS LIMITED: For situation where all target units met and remaining WETH, uniformly raise targets by same
-     * percentage in order to allow further trading. Can be called multiple times if necessary, increase should be
-     * small in order to reduce tracking error.
+     * ACCESS LIMITED: For situation where all target units met and remaining WETH, uniformly raise targets by same percentage by applying
+     * to logged positionMultiplier in RebalanceInfo struct. in order to allow further trading. Can be called multiple times if necessary,
+     * targets are increased by amount specified by raiseAssetTargetsPercentage as set by manager.
```
Comment (cgewecke) [inline contracts/protocol/modules/GeneralIndexModule.sol:322]: In the spec it says you might call this multiple times .. "slowly raising the target in order to limit tracking error". 

Would it be possible to add a short explanation of that context / strategy?

**Review by cgewecke — COMMENTED** [review]
These comments are so great 😄 

Super clear and detailed.

File: contracts/protocol/modules/GeneralIndexModule.sol [hunk 2]
```diff
@@ -40,12 +40,17 @@ import { Uint256ArrayUtils } from "../../lib/Uint256ArrayUtils.sol";
...
  *
- * Smart contract that facilitates rebalances for indices. Manager can set target unit amounts, max trade sizes, the
- * exchange to trade on, and the cool down period between trades (on a per asset basis). 
+ * Smart contract that facilitates rebalances for indices. Manager can update allocation by calling startRebalance().
+ * There is no "end" to a rebalance, however once there are no more tokens to sell the rebalance is effectively over
+ * until the manager calls startRebalance() again with a new allocation. Once a new allocation is passed in, allowed 
+ * traders can submit rebalance transactions by calling trade() and specifying the component they wish to rebalance.
+ * All parameterizations for a trade are set by the manager ahead of time, including max trade size, coolOffPeriod bet-
+ * ween trades, and exchange to trade on. Once a component's target allocation is met any further attempted trades of
+ * that component will revert.
```
Comment (felix2feng) [inline contracts/protocol/modules/GeneralIndexModule.sol:49]: One good thing to add is the assumption that all trades use WETH as a bridge asset. And that there is functions to handle remaining WETH and when the WETH amount gained is more than expected

**Review by cgewecke — APPROVED** [review]
💯

**Pull request merged**